### PR TITLE
Allow replace metadata.name and metadata.namespace for single yaml structure manifest

### PR DESCRIPTION
**Pull request by serngawy**
Allow replace metadata.name and metadata.namespace for single yaml structure manifest

Signed-off-by: melserngawy <melserng@redhat.com>

### CODE REVIEW

File: internal/utils_test.go [hunk 1]
```diff
@@ -253,6 +253,127 @@ data:
...
+	metadata, ok := objDef["metadata"].(map[string]interface{})
+	if !ok {
+		t.Fatal("The metadata field is an invalid format")
+	}
+
+	name, ok := metadata["name"].(string)
+	if !ok {
+		t.Fatal("The metadata.name field is an invalid format")
+	}
+	assertReflectEqual(t, name, "patch-configmap")
```
Comment (mprahl) [inline internal/utils_test.go:329]:
```suggestion
	assertEqual(t, name, "patch-configmap")
```

File: internal/utils_test.go [hunk 2]
```diff
@@ -253,6 +253,127 @@ data:
...
+
+	namespace, ok := metadata["namespace"].(string)
+	if !ok {
+		t.Fatal("The metadata.namespace field is an invalid format")
+	}
+	assertReflectEqual(t, namespace, "patch-namespace")
```
Comment (mprahl) [inline internal/utils_test.go:335]:
```suggestion
	assertEqual(t, namespace, "patch-namespace")
```

File: internal/utils_test.go [hunk 3]
```diff
@@ -253,6 +253,127 @@ data:
...
+	data, ok := objDef["data"].(map[string]interface{})
+	if !ok {
+		t.Fatal("The data field is an invalid format")
+	}
+
+	image, ok := data["image"].(string)
+	if !ok {
+		t.Fatal("The data.image field is an invalid format")
+	}
+	assertReflectEqual(t, image, "quay.io/potatos2")
```
Comment (mprahl) [inline internal/utils_test.go:346]:
```suggestion
	assertEqual(t, image, "quay.io/potatos2")
```

File: internal/utils_test.go [hunk 4]
```diff
@@ -253,6 +253,127 @@ data:
...
+
+	// Test not allowing replace metadata.name or metadata.namespace with multi yaml structures in same manifest
```
Comment (mprahl) [inline internal/utils_test.go:348]: Could you please move this section to its own test?

File: docs/policygenerator-reference.yaml [hunk 5]
```diff
@@ -76,9 +76,11 @@ policies:
             kind: ""
             metadata:
               # Optional: Only required when there are multiple manifests in the path.
+              # Optional: replace metadata.name allowed only for single yaml structure manifest
```
Comment (mprahl) [inline docs/policygenerator-reference.yaml:79]:
```suggestion
              # Optional: Replace metadata.name is only allowed for a single YAML structure manifest.
```

File: docs/policygenerator-reference.yaml [hunk 6]
```diff
@@ -76,9 +76,11 @@ policies:
...
               name: ""
               # Optional: Only required when there are multiple manifests in the path and it's a
               # manifest to a namespaced resource.
+              # Optional: replace metadata.namespace allowed only for single yaml structure manifest
```
Comment (mprahl) [inline docs/policygenerator-reference.yaml:83]:
```suggestion
              # Optional: Replace metadata.namespace is only allowed for a single YAML structure manifest.
```

File: internal/utils.go [hunk 7]
```diff
@@ -47,10 +48,38 @@ func getManifests(policyConf *types.PolicyConfig) ([]map[string]interface{}, err
...
+			if len(*manifestFile) == 0 {
+				continue
+			}
+			// Allowing replace the original manifest metadata.name and/or metadata.namespace if it is a single
+			// yaml structure in the manifest path
+			if len(*manifestFile) == 1 && len(manifest.Patches) == 1 {
+				patchMetadata, ok := manifest.Patches[0]["metadata"].(map[string]interface{})
+				if ok {
+					metadata, ok := (*manifestFile)[0]["metadata"].(map[string]interface{})
+					if ok {
```
Comment (mprahl) [inline internal/utils.go:66]: Optional: This logic gets pretty nested in a bunch of if statements. How about something like this to make it a little easier to read?
```suggestion
				patchMetadata, patchOk := manifest.Patches[0]["metadata"].(map[string]interface{})
				metadata, ok := (*manifestFile)[0]["metadata"].(map[string]interface{})
				if patchOk && ok {
```

File: internal/utils_test.go (hunk 1 above)
Comment (serngawy) [reply internal/utils_test.go]: done

File: docs/policygenerator-reference.yaml (hunk 5 above)
Comment (serngawy) [reply docs/policygenerator-reference.yaml]: done

**Pull request merged**